langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/InternalQianfanHelper.java

```diff
@@ -179,10 +179,18 @@ static String getSystemMessage(List<ChatMessage> messages) {
 
     }
     public static List<Message> toOpenAiMessages(List<ChatMessage> messages) {
-        return messages.stream()
+         List<Message> aiMessages = messages.stream()
                 .filter(chatMessage -> !(chatMessage instanceof SystemMessage))
                 .map(InternalQianfanHelper::toQianfanMessage)
                 .collect(toList());
+
+         // ensure the length of messages is odd
+        if(aiMessages.size() == 0 || aiMessages.size() % 2 == 1){
```

Comment:
What is the reason for this "must be odd" requirement?

Reply:
If the length of messages is even, the Qianfan model will return an error code 336006. For reference, please see the link:https://cloud.baidu.com/doc/WENXINWORKSHOP/s/tlmyncueh

Reply:
I understand that the error will be returned. But what is behind this requirement? Why can't the length be even?

Reply:
I don't know either. I'm just a user, not a developer of the Qianfan model.